File: tttyoung/week_23/boj_14425_문자열집합.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
s = set() #집합사용용
count = 0 #개수수

for _ in range(n): #주어진 s집합 입력
    s.add(input()) 
for _ in range(m): #m개의 문자열을 돌면서 이 중에 입력된 s집합과 공통된것이 있는지 확인
    x = input()
    if x in s:
        count+=1 #공통된 값일때 count+1

print(count)

File: miggule2/week25/BOJ_14425_문자열집합.java
package week25;

import java.util.*;
import java.io.*;

public class BOJ_14425_문자열집합 {
    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] num = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int n = num[0];
        int m = num[1];

        HashSet<String> set = new HashSet<>();
        for(int i = 0; i < n; i++){
            set.add(br.readLine());
        }

        int count = 0;
        for(int i = 0; i < m; i++){
            if(set.contains(br.readLine())) count++;
        }

        System.out.println(count);
    }
}
